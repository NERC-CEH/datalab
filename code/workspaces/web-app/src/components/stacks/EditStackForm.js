import React from 'react';
import PropTypes from 'prop-types';
import { Field, reduxForm } from 'redux-form';
import validate from 'validate.js';
import { renderTextArea, renderTextField, UpdateFormControls } from '../common/form/controls';

const validationConstraints = {
  displayName: {
    presence: true,
  },
  description: {
    presence: true,
  },
};

const commonFormProps = {
  onCancel: PropTypes.func.isRequired,
};

const EditStackForm = ({
  handleSubmit, reset, pristine, onCancel,
}) => (
  <form onSubmit={handleSubmit}>
    <Field
      name="displayName"
      label="Display Name"
      component={renderTextField}
    />
    <Field
      name="description"
      label="Description"
      component={renderTextArea}
    />
    <UpdateFormControls onClearChanges={reset} onCancel={onCancel} pristine={pristine} />
  </form>
);

EditStackForm.propTypes = {
  ...commonFormProps,
  handleSubmit: PropTypes.func.isRequired,
  reset: PropTypes.func.isRequired,
  pristine: PropTypes.bool.isRequired,
};

const EditStackReduxForm = reduxForm({
  form: 'editStackDetails',
  validate: values => validate(values, validationConstraints, { format: 'reduxForm' }),
})(EditStackForm);

EditStackReduxForm.propTypes = {
  ...commonFormProps,
  onSubmit: PropTypes.func.isRequired,
  initialValues: PropTypes.shape({
    displayName: PropTypes.string,
    description: PropTypes.string,
  }),
};

export { EditStackForm as PureEditStackForm };

export default EditStackReduxForm;
