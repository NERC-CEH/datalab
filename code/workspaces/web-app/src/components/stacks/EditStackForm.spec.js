import React from 'react';
import { shallow } from 'enzyme';
import { PureEditStackForm } from './EditStackForm';

describe('EditStackForm', () => {
  const shallowRender = () => shallow(
    <PureEditStackForm
      handleSubmit={jest.fn().mockName('handleSubmit')}
      reset={jest.fn().mockName('reset')}
      pristine={true}
      onCancel={jest.fn().mockName('onCancel')}
    />,
  );

  it('renders to match snapshot', () => {
    const pristine = true;
    expect(shallowRender(pristine)).toMatchSnapshot();
  });
});
