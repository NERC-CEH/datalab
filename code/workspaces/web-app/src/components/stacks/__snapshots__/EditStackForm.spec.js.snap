// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditStackForm renders to match snapshot 1`] = `
<form
  onSubmit={[MockFunction handleSubmit]}
>
  <Field
    component={[Function]}
    label="Display Name"
    name="displayName"
  />
  <Field
    component={[Function]}
    label="Description"
    name="description"
  />
  <UpdateFormControls
    onCancel={[MockFunction onCancel]}
    onClearChanges={[MockFunction reset]}
    pristine={true}
  />
</form>
`;
